awx/api/serializers.py

```diff
@@ -4230,6 +4235,67 @@ def get_summary_fields(self, obj):
         d['recent_notifications'] = self._recent_notifications(obj)
         return d
 
+    def validate_messages(self, messages):
+        if messages is None:
+            return None
+
+        error_list = []
+        collected_messages = []
+
+        if not isinstance(messages, dict):
+            error_list.append(_("Expected dict for 'messages' field, found {}".format(type(messages))))
+        else:
+            for event in messages:
+                if event not in ['started', 'success', 'error']:
+                    error_list.append(_("Event '{}' invalid, must be one of 'started', 'success', or 'error'").format(event))
```

Comment:
As a knee jerk reaction, I don't _love_ reflecting the users bad input. But I don't see any real problem in this case.

```
{
    "name": "",
    "description": "",
    "organization": null,
    "notification_type": "blah",
    "notification_configuration": {},
    "messages": {
        "started": {
            "message": "{{ job_friendly_name }} #{{ job.id }} '{{ job.name }}' {{ job.status }}: {{ url }}",
            "body": ""
        },
        "success": {
            "message": "{{ job_friendly_name }} #{{ job.id }} '{{ job.name }}' {{ job.status }}: {{ url }}",
            "body": ""
        },
        "error": {
            "message": "{{ job_friendly_name }} #{{ job.id }} '{{ job.name }}' {{ job.status }}: {{ url }}",
            "body": ""
        },
        "sniggsnort\"></span><a href='blah.com'>hahah</a>": {}
    }
}
```

```
{
    "name": [
        "This field may not be blank."
    ],
    "notification_type": [
        "\"blah\" is not a valid choice."
    ],
    "messages": [
        "Event 'sniggsnort\"></span><a href='blah.com'>hahah</a>' invalid, must be one of 'started', 'success', or 'error'"
    ]
}
```